### PR TITLE
Added BackgroundModeHandler, which knows if the app is in foreground

### PR DESCRIPTION
**Pull request by d3xter**

### CODE REVIEW

File: app/src/main/java/org/blitzortung/android/app/BackgroundModeHandler.kt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import android.content.Context
+import android.os.Handler
+import android.util.Log
+import org.blitzortung.android.app.event.BackgroundModeEvent
+import org.blitzortung.android.protocol.ConsumerContainer
+
+class BackgroundModeHandler(private val context: Context)  {
+    val handler = Handler(context.mainLooper)
+
+    private var runable: Runnable? = null
```
Comment (wuan) [inline app/src/main/java/org/blitzortung/android/app/BackgroundModeHandler.kt]: just the typo: runable -> runnable

File: app/src/main/java/org/blitzortung/android/app/BackgroundModeHandler.kt [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            runable = Runnable {
+                sendUpdates(isInBackground)
+
+                runable = null
+            }
+
+            handler.postDelayed(runable, 500)
+        } else {
+            if(runable is Runnable) {
+                handler.removeCallbacks(runable)
```
Comment (wuan) [inline app/src/main/java/org/blitzortung/android/app/BackgroundModeHandler.kt]: should `runnable = null` be added?

File: app/src/main/java/org/blitzortung/android/app/event/BackgroundModeEvent.kt [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package org.blitzortung.android.app.event
+
+import org.blitzortung.android.protocol.Event
+
+class BackgroundModeEvent(val isInBackground: Boolean): Event
```
Comment (wuan) [inline app/src/main/java/org/blitzortung/android/app/event/BackgroundModeEvent.kt]: use `data class` to get to String and equals etc.?

File: app/src/main/java/org/blitzortung/android/app/BackgroundModeHandler.kt (hunk 2 above)
Comment (d3xter) [inline app/src/main/java/org/blitzortung/android/app/BackgroundModeHandler.kt]: what do you mean?

Comment (wuan) [inline app/src/main/java/org/blitzortung/android/app/BackgroundModeHandler.kt]: When the runnable is executed, it sets the field runnable to null. This is not the case when the runnable is removed before it is executed. In this case the field runnable is still set.

Comment (d3xter) [inline app/src/main/java/org/blitzortung/android/app/BackgroundModeHandler.kt]: You're right, I've changed at now.

Thx :)

**Pull request merged**